tools/tools.go

```diff
@@ -211,9 +211,11 @@ func (p *Parser) findArguments() (map[string]any, int) {
 			// check if the object keys are valid tool properties
 			// TODO (jmorganca): check only sets of properties that
 			// go together instead of the entire set
-			for _, prop := range p.properties {
-				if _, exists := v[prop]; exists {
-					return v
+			for _, t := range p.tools {
+				for p := range t.Function.Parameters.Properties {
+					if _, exists := v[p]; exists {
+						return v
```

Comment:
I might be reading this wrong but is multi property case handled?

Reply:
Yes! This should be handled now. It makes sure all keys are in the list of properties for the tool call